docs/source/core/embedding.rst

```diff
@@ -171,6 +171,58 @@ To obtain an api key, visit: https://octo.ai/docs/getting-started/how-to-create-
     query = "This is the query"
     query_embedding = embedding_encoder.embed_query(query=query)
 
+    [print(e.embeddings, e) for e in elements]
+    print(query_embedding, query)
+    print(embedding_encoder.is_unit_vector(), embedding_encoder.num_of_dimensions())
+
+``VertexAIEmbeddingEncoder``
+--------------------------
+
+The ``VertexAIEmbeddingEncoder`` class connects to the GCP VertexAI to obtain embeddings for pieces of text.
+
+``embed_documents`` will receive a list of Elements, and return an updated list which
+includes the ``embeddings`` attribute for each Element.
+
+``embed_query`` will receive a query as a string, and return a list of floats which is the
+embedding vector for the given query string.
+
+``num_of_dimensions`` is a metadata property that denotes the number of dimensions in any
+embedding vector obtained via this class.
+
+``is_unit_vector`` is a metadata property that denotes if embedding vectors obtained via
+this class are unit vectors.
+
+The following code block shows an example of how to use ``VertexAIEmbeddingEncoder``. You will
+see the updated elements list (with the ``embeddings`` attribute included for each element),
+the embedding vector for the query string, and some metadata properties about the embedding model.
+
+You will need to have the credentials configured for your environment (gcloud, workload identity,
+etc…) or you'll need to store the path to a service account JSON file as the
+GOOGLE_APPLICATION_CREDENTIALS environment variable to be able to run this example.
+For more information: https://python.langchain.com/docs/integrations/text_embedding/google_vertex_ai_palm
+
+# Or, you can pass the json content of your API key to the VertexAIEmbeddingConfig
+# like this: VertexAIEmbeddingConfig(api_key_json=GOOGLE_APPLICATION_CREDENTIALS_JSON_CONTENT)
+# this will create a file in the working directory with the content of the json, and set the
+# GOOGLE_APPLICATION_CREDENTIALS environment variable to the path of the file.
+
+.. code:: python
+
+    import os
+
+    from unstructured.documents.elements import Text
+    from unstructured.embed.vertexai import VertexAIEmbeddingConfig, VertexAIEmbeddingEncoder
+
+    embedding_encoder = VertexAIEmbeddingEncoder(
+        config=VertexAIEmbeddingConfig()
```

Comment:
This is asking for api_key

Reply:
[working example in docs with comments updated](https://github.com/Unstructured-IO/unstructured/pull/2693/commits/7587fd3c90ef958d3e08b2033b18b2bb6fa49389)